spring-cloud-kubernetes-fabric8-discovery/src/main/java/org/springframework/cloud/kubernetes/fabric8/discovery/KubernetesCatalogWatch.java

```diff
@@ -84,6 +84,12 @@ public void catalogServicesWatch() {
 
 	@PostConstruct
 	void postConstruct() {
+		stateGenerator = stateGenerator();
+	}
+
+	Function<Fabric8CatalogWatchContext, List<EndpointNameAndNamespace>> stateGenerator() {
```

Comment:
This eases the testing _a lot_, as I can create a Spy now from this class and only mock the result of this method.